### PR TITLE
Fix wallet startup

### PR DESCRIPTION
**Pull request by delta1**
## Description

If we can't get base node chain metadata when fetching balance, then don't fail with an error 

## Motivation and Context

Fixes the wallet failing at startup 

## How Has This Been Tested?

Manually 

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by StriderDM — APPROVED** [review]
LGTM

File: base_layer/wallet/src/output_manager_service/service.rs [hunk 1]
```diff
@@ -264,12 +264,11 @@ where TBackend: OutputManagerBackend + 'static
...
-                    .base_node_service
-                    .get_connected_base_node_state()
-                    .await?
-                    .height_of_longest_chain();
-                self.get_balance(Some(tip)).await.map(OutputManagerResponse::Balance)
+                let tip_or_none = match self.base_node_service.get_connected_base_node_state().await {
+                    Ok(metadata) => Some(metadata.height_of_longest_chain()),
+                    Err(_) => None,
+                };
+                self.get_balance(tip_or_none).await.map(OutputManagerResponse::Balance)
```
Comment (philipr-za) [inline base_layer/wallet/src/output_manager_service/service.rs:271]: A neater Rust way to do this:
```Rust
self.get_balance(self.base_node_service.get_connected_base_node_state().await.map(|r| r.height_of_longest_chain).ok()).await.map(OutputManagerResponse::Balance)
```

Comment (delta1) [reply base_layer/wallet/src/output_manager_service/service.rs:271]: ~~Yeah I was just wondering about `.ok()` - but you still need to call `height_of_longest_chain()` ?~~

Oh very nice - so map will leave the Error and then .ok() will convert it to None

**Review by philipr-za — APPROVED** [review]

File: base_layer/wallet/src/output_manager_service/service.rs (hunk 1 above)
Comment (philipr-za) [reply base_layer/wallet/src/output_manager_service/service.rs:271]: Soz, left out the map.

**Pull request merged**